shared/repository/lib/api/documents/ContactRepository.mjs

```diff
@@ -0,0 +1,150 @@
+        const params = [userId, trackedPhone];
+        const result = await this.db.queryAsync(sql, params).catch((err) => {
+            throw new RepoError(err);
+        });
+        return result && result.rows[0];
+    }
+
+    /**
+     * @method
+     * @param {number} userId - user identifier
+     * @param {boolean} tracked - flag that indicates is the contact tracked
+     * @param {string} trackedPhone - contact added by user that should be tracked
+     * @param {string} sessionId - identifier of session that tracks this contact
+     * @param {boolean} withSession - flag that indicates does contact have session
+     * @return {Promise<Object>} returns data saved in DB
+     */
+    async read({ userId, tracked, trackedPhone, sessionId, withSession }) {
+        let sql = 'SELECT * FROM tracked_phones';
+        let params = '';
+
+        if (typeof userId === 'number') {
+            params += ` user_id = ${userId}`;
+        }
+
+        if (typeof tracked === 'boolean') {
+            params !== '' ? (params += ` AND tracked = ${tracked}`) : (params += ` tracked = ${tracked}`);
+        }
+
+        if (typeof trackedPhone === 'string') {
+            params !== ''
+                ? (params += ` AND tracked_phone = ${trackedPhone}`)
+                : (params += ` tracked_phone = ${trackedPhone}`);
+        }
+
+        if (typeof sessionId === 'string') {
+            params !== '' ? (params += ` AND session_id = ${sessionId}`) : (params += ` session_id = ${sessionId}`);
+        }
+
+        if (typeof withSession === 'boolean') {
+            if (withSession) {
+                params !== '' ? (params += ' AND session_id IS NOT NULL') : (params += ' session_id IS NOT NULL');
+            } else {
+                params !== '' ? (params += ' AND session_id IS NULL') : (params += ' session_id IS NULL');
+            }
+        }
+
+        if (params !== '') sql += ' WHERE' + params;
+
+        const result = await this.db.queryAsync(sql).catch((err) => {
+            throw new RepoError(err);
+        });
+        return result && result.rows;
+    }
+
+    /**
+     * @method
+     * @param {Number} userId - user identifier
+     * @param {String} trackedPhone - contact added by user that should be tracked
+     * @returns {Promise<Object>}
+     */
+    async save({ userId, trackedPhone }) {
+        const result = await this.db
+            .queryAsync('INSERT INTO tracked_phones (user_id, tracked_phone) VALUES ($1, $2)', [userId, trackedPhone])
+            .catch((err) => {
+                throw new RepoError(err);
+            });
+        return result;
+    }
+
+    /**
+     * @method
+     * @param {Number} userId - user identifier
+     * @param {String} trackedPhone - contact added by user that should be tracked
+     * @param {Boolean} tracked - The flag that indicates is the contact tracked
+     * @returns {Promise<Object>}
+     */
+
+    async update({ userId, trackedPhone, tracked }) {
```

Comment:
I thought that we agreed that it should be all properties list

Reply:
done